ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -70,125 +56,26 @@ export class ImportedDocuments extends React.Component {
-    const { taskId } = this.props;
-    if (!importCompleted) {
-      const nextEntry = _get(data, 'loaded_entries', 0);
-      const response = await importerApi.check(taskId, nextEntry);
-      if (response.data.status !== 'RUNNING') {
-        this.setState({
-          importCompleted: true,
-          isLoading: false,
-          data: response.data,
-        });
-      } else {
-        this.setState({
-          isLoading: true,
-          data: response.data,
-        });
-      }
-      this.filteredRecords();
-      this.calculateMode();
-      this.calculateStatus();
-      this.calculateStatistics(response.data);
-    } else {
-      this.intervalId && clearInterval(this.intervalId);
-    }
-  };
-
-  calculateMode = () => {
-    const { data } = this.state;
-    switch (data.mode) {
-      case 'IMPORT':
-        this.mode = (
-          <Label color="blue" basic>
-            Import
-          </Label>
-        );
-        break;
-      case 'DELETE':
-        this.mode = (
-          <Label color="red" basic>
-            Delete
-          </Label>
-        );
-        break;
-      case 'PREVIEW_IMPORT':
-        this.mode = (
-          <Label color="teal" basic>
-            Preview (import)
-          </Label>
-        );
-        break;
-      case 'PREVIEW_DELETE':
-        this.mode = (
-          <Label color="teal" basic>
-            Preview (delete)
-          </Label>
-        );
-        break;
-      case 'ERROR':
-        this.mode = (
-          <Label color="red" basic>
-            Error
-          </Label>
-        );
-        break;
-      default:
-        throw Error('There was a problem processing the mode.');
-    }
-  };
+  componentDidUpdate(prevProps) {
+    const { data } = this.props;
 
-  statusLabel = () => {
-    const { data, isLoading } = this.state;
-    if (!data) return <Label color="grey">Fetching status</Label>;
-    if (isLoading)
-      return (
-        <Label color="blue">
-          <Icon loading name="circle notch" /> Importing literature
-        </Label>
-      );
-    switch (data.status) {
-      case 'SUCCEEDED':
-        return <Label color="green">Import successfull</Label>;
-      case 'CANCELLED':
-        return <Label color="yellow">Import cancelled</Label>;
-      default:
-        return <Label color="red">Import failed</Label>;
+    if (data && data !== prevProps.data) {
+      this.filterRecords();
+      this.calculateStatistics(data);
     }
-  };
-
-  calculateStatus = () => {
-    this.status = this.statusLabel();
-  };
+  }
 
   calculateStatistics = data => {
-    const { statistics } = this.state;
+    const { statistics, lastIndex } = this.state;
     let stats = _cloneDeep(statistics);
     const records = data.records;
-    const newRecords = records.slice(this.lastIndex, records.length);
+    const newRecords = records.slice(lastIndex, records.length);
 
-    this.lastIndex += newRecords.length;
+    this.setState({ lastIndex: lastIndex + newRecords.length });
```

Comment:
I would move this down to the final `setState` before returning, to have only one `setState`: normally it should not immediately re-render, but @Pineirin observed that it was re-rendering on each `setState`.